invenio_app_rdm/records_ui/views/decorators.py

```diff
@@ -158,6 +158,11 @@ def get_record_file_content():
                 item = get_record_file_content()
         else:
             item = get_record_file_content()
+
+        # note: the file service returns None if the record doesn't have the file
+        if item is None:
```

Comment:
moderate: I agree with @fenekku on raising at the service level, and I'd even create a custom exception (if doesn't exist already). This would allow a more fine grained error handling/reporting. `KeyError` is a bit too generic.  I think this might also allow to simplify the logic above.

Reply:
Perfect, then I'll rework it to match this!

Oopsy, sorry regarding the PR triage!
We're getting spammed with these error messages lately and I was a bit distracted, so I didn't pay the attention to the process I should've :sweat_smile:

Reply:
No worries about the triaging, I think it's a good discussion point to bring up with the group. Maybe the triage group will have some good thoughts on this.